server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3808,9 +3815,29 @@ public List<String> getActiveOperations() {
 
     private static AsyncIOProcessor<Translog.Location> createTranslogSyncProcessor(
         Logger logger,
-        ThreadContext threadContext,
-        Supplier<Engine> engineSupplier
+        ThreadPool threadPool,
+        Supplier<Engine> engineSupplier,
+        boolean bufferAsyncIoProcessor,
+        TimeValue bufferInterval
     ) {
+        ThreadContext threadContext = threadPool.getThreadContext();
+        if (bufferAsyncIoProcessor) {
+            return new BufferedAsyncIOProcessor<>(logger, 102400, threadContext, threadPool, bufferInterval) {
+                @Override
+                protected void write(List<Tuple<Translog.Location, Consumer<Exception>>> candidates) throws IOException {
```

Comment:
implementation of `write` is same for AsyncIOProcesseor and BufferedAsyncIOProcessor. Can we abstract it out?

Reply:
Incorporated.